### PR TITLE
run only related tests in the pre-commit hook

### PR DESCRIPTION
**Pull request by kevinbarabash**
We could make the check even faster if we skipped the build step, but then we'd have to modify the module mapper settings in the jest config.  TBH, I'm not sure how much value we get from testing the built versions of the packages.

Test Plan:
- `yarn test:fast`, see that only the commonjs modules are built and that only a subset of the tests run
- `git commit -a -m ...`, see that the same output as for `yarn test:fast`

### CODE REVIEW

File: packages/wonder-blocks-button/components/button.js [hunk 1]
```diff
@@ -205,3 +205,5 @@ export default class Button extends React.Component<SharedProps> {
         );
     }
 }
+
+// trivial change to trigger tests
```
Comment (kevinbarabash) [inline packages/wonder-blocks-button/components/button.js]: I'll revert this change once people are happy with the rest of this PR.

File: package.json [hunk 2]
```diff
@@ -30,6 +30,7 @@
     "start": "concurrently --kill-others-on-fail \"yarn run styleguidist\" \"webpack --config build-settings/webpack.config.js -w\" \"rollup -c build-settings/rollup.config.js -w\"",
     "test:common": "yarn run build:all && yarn run lint && yarn run alex && yarn run gen-snapshot-tests && yarn run flow",
     "test:coverage": "yarn run test:common && yarn run jest --coverage",
+    "test:fast": "yarn build:cjs && yarn jest --findRelatedTests `git diff --cached --name-only --diff-filter=ACM | grep '\\.js$'`",
```
[the remark this answers is not in the document]
Comment (jeresig) [reply package.json]: I feel like the likelihood of having an "R" match but no other of ACM match feels very unlikely to me.

**Comment by kevinbarabash** [comment]
> It feels like we should probably be using husky and lint-staged to manage this stuff: https://github.com/okonet/lint-staged but this is fine for now.

lint-staged looks pretty nice.  What's the benefit of `husky` over the existing pre-commit setup?

**Comment by jeresig** [comment]
@kevinbarabash Husky allows you to hook in to any Git hook: https://git-scm.com/docs/githooks so we could change it to run on push, like we do currently for webapp.

**Comment by kevinbarabash** [comment]
@jeresig husky is a bit more awkward when you want to run multiple commands, see https://github.com/typicode/husky#multiple-commands.  Another option is using https://github.com/dflourusso/pre-push which works similarly to `pre-commit`.

**Pull request merged**